core/src/main/java/com/novoda/noplayer/model/AudioTracks.java

```diff
@@ -0,0 +1,78 @@
+package com.novoda.noplayer.model;
+
+import com.novoda.noplayer.internal.exoplayer.mediasource.AudioTrackType;
+import com.novoda.utils.Optional;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+public class AudioTracks implements Iterable<PlayerAudioTrack> {
+
+    private final List<PlayerAudioTrack> audioTracks;
+
+    public static AudioTracks from(List<PlayerAudioTrack> audioTracks) {
+        return new AudioTracks(Collections.unmodifiableList(audioTracks));
+    }
+
+    private AudioTracks(List<PlayerAudioTrack> audioTracks) {
+        this.audioTracks = audioTracks;
+    }
+
+    public boolean containsPlayerAudioTrackWith(AudioTrackType trackType) {
```

Comment:
nope I follow that it would work technically, I meant it's not semantically correct (which is important for first class collection which is all just semantics) for a collection of type X to return true for a method `contains(TypeY)` because it doesn't contain TypeY, it contains things that may be of Type Y.